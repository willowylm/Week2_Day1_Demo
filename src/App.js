import React, { Component } from 'react';

import PropTypes from "prop-types"
import './App.css';
import Header from './Header'

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      desc:"react demo",
      appName:"react appName",
      list: []
    };
    this.onClick=this.changeAppDesc.bind(this);
  }

   componentDidUpdate(){
    console.log(this.componentDidUpdate);
   }

   changeAppnameDesc=(()=>{
    this.setState({
      desc:"react redux"
    });
  })
  
  displayJson = () => {
    fetch('https://api.github.com/search/repositories?q=topic:ruby+topic:rails')
      .then((response)=> {
        return response.json();
      })
      .then((myJson)=> {
        console.log(myJson);
         this.setState({list: myJson.items.slice(0, 10)})
      }).catch((error) => {
        // console.log(error)
      })
  }


  changeAppDesc(){
    this.setState({
      desc:"react demo 2018"
    });

  }
  render() {
    return (
      <div className="App">
       <Header appName= {this.state.appName} />
        <p className="App-intro">
          {/* To get started, edit <code>src/App.js</code> and save to reload. */}
          {this.state.desc}
        </p>
        <button className='btn1' onClick={this.displayJson}> displayjson</button>
        <button onClick={this.onClick}> change desc</button>
        {
          this.state.list.map(item => {
            return <li key={item.id}>{item.url},{item.description}</li>
                 
          })
        }
      </div>
  );
  }
}

App.propTypes = {
  hello: PropTypes.string.isRequired
};

export default App;
