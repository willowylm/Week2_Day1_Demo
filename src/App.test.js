import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App hello="aa"/>, div);
  ReactDOM.unmountComponentAtNode(div);
});



it('renders App correctly',()=>{
const tree = renderer.create(<App hello ="hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
});