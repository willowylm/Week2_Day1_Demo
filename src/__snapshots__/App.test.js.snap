// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      react appName
    </h1>
  </header>
  <p
    className="App-intro"
  >
    react demo
  </p>
  <button
    className="btn1"
    onClick={[Function]}
  >
     displayjson
  </button>
  <button
    onClick={[Function]}
  >
     change desc
  </button>
</div>
`;
